pkg/subscraping/sources/sitedossier/sitedossier.go

```diff
@@ -29,13 +30,19 @@ func (a *agent) enumerate(ctx context.Context, baseURL string) error {
 			return nil
 		default:
 			resp, err := a.session.SimpleGet(ctx, baseURL)
-			if err != nil {
+
+			isnotfound := resp != nil && resp.StatusCode == http.StatusNotFound
```

Comment:
We shuld still read response body even if we get a 404 response since go will not reuse that connection until and unless we read all data from it. Can we change this to allow reading response and then returning?

Reply:
Ouch, my mistake, sure.